packages/react-ui-core/src/Gmap/FreeDrawLayer.js

```diff
@@ -0,0 +1,202 @@
+import React, { PureComponent } from 'react'
+import PropTypes from 'prop-types'
+import autobind from 'autobind-decorator'
+import themed from 'react-themed'
+import { EVENTS } from './utils/const'
+import { setupEvents, removeEvent } from './utils/mapEventHelpers'
+
+const MAP_CONTROLS_ENABLED = {
+  draggable: true,
+  zoomControl: true,
+  scrollwheel: true,
+  disableDoubleClickZoom: true,
+}
+
+const MAP_CONTROLS_DISABLED = {
+  draggable: false,
+  zoomControl: false,
+  scrollwheel: false,
+  disableDoubleClickZoom: false,
+}
+@themed(['Gmap_FreeDrawTootip'], { pure: true })
+export default class FreeDrawLayer extends PureComponent {
+  static propTypes = {
+    map: PropTypes.object,
+    onMapDrawStart: PropTypes.func,
+    onMapDrawEnd: PropTypes.func,
+    theme: PropTypes.object,
+    mapControls: PropTypes.object,
+    shapes: PropTypes.object,
+    multipleShapes: PropTypes.bool,
+    disabled: PropTypes.bool,
+    dataStyle: PropTypes.object,
+  }
+
+  static defaultProps = {
+    theme: {},
+    map: {},
+    multipleShapes: false,
+    disabled: false,
+    shapes: {},
+    dataStyle: {},
+  }
+
+  constructor(props) {
+    super(props)
+    this.state = {
+      drawing: false,
+    }
+    this.polygons = Object.keys(this.props.shapes).length ? [this.addShapeToGMaps(
+      this.formatLongLatToGmapsCoordinates(
+        this.props.shapes
+      )[0]
+    )] : []
+  }
+
+  componentDidUpdate(prevProps) {
+    if (this.props.shapes !== prevProps.shapes) {
+      this.clearAllShapes()
+      this.addShape(
+        this.formatLongLatToGmapsCoordinates(
+          this.props.shapes
+        )[0]
+      )
+    }
+  }
+
+  getPolygonCoords(poly) {
+    const len = poly.getPath().getLength()
+    const shape = []
+
+    for (let i = 0; i < len; i += 1) {
+      const path = poly.getPath().getAt(i)
+      shape.push([path.lng(), path.lat()])
+    }
+    return shape
+  }
+
+  enableMapControls() {
+    const { map, mapControls } = this.props
+    map.setOptions({
+      ...MAP_CONTROLS_ENABLED,
+      ...mapControls,
+    })
+    this.setState({
+      drawing: false,
+    })
+  }
+
+  disableMapControls() {
+    const { map, mapControls } = this.props
+    map.setOptions({
+      ...MAP_CONTROLS_DISABLED,
+      ...mapControls,
```

Comment:
That doesn't work... What if someone passed in `zoomControl: true` in the mapControls prop???